pkg/clustermesh/common/remote_cluster.go

```diff
@@ -132,6 +135,19 @@ func (rc *remoteCluster) releaseOldConnection() {
 }
 
 func (rc *remoteCluster) restartRemoteConnection() {
+	connectionEstablished := false
+	// Applicable to kvstoremesh only:
+	// We execute a goroutine to implement a timeout on connection.
+	// If the timeout has been reached and a connection hasn't been established, we
+	// remove the cluster from the SyncState to prevent it from blocking readiness forever.
+	go func() {
+		<-time.After(option.Config.KVstoreConnectivityTimeout)
```

Comment:
hmm, I'm unsure if we shouldn't have a separate parameter for it like `KVStoreMeshConnectivityTimeout`. Resuing KVstore one might be confusing.

Reply:
I would probably also explicitly add it to helm as a parameter - once we have HA, users might want to increase it also depending on clustermesh size.

Reply:
Yeah I was also back and forth on this, since the option does seem applicable but is also used in a different context elsewhere. I'll make a new option for this